### PR TITLE
fix: validate bids order

### PR DESCRIPTION
**Pull request by nikkaroraa**
# Task:
Fix "validate bids" order

## Description
Using Promise.all doesn't guarantee the order in which the promises are going to be resolved. Whichever gets the response first, gets resolved first. As a result of that, the value of filledAmt was getting increased in a random order. 

We made this change to incorporate async nonce checks.

Fix is to separately get the nonce check and then synchronously filter the bids in the order that they are supposed to.

Fixes ENG-2045

## Type of change
- [ ] New feature
- [x] Bug fix
- [ ] Testing code
- [ ] Document update or config files

### CODE REVIEW

**Review by alpinechicken — APPROVED** [review]
lgtm

File: utils/auction.ts [hunk 1]
```diff
@@ -65,6 +65,19 @@ export const getUniqueTraders = (bids: Array<Bid>) => {
...
+  const nonceCheckPromises = bids.map(async bid => {
+    const isNonceUsed = await crabV2Contract.nonces(bid.bidder, bid.order.nonce)
+    return { bidder: bid.bidder, nonce: bid.order.nonce, isNonceUsed }
+  })
+
+  const nonceResults = await Promise.all(nonceCheckPromises)
+  return nonceResults.reduce((acc, { bidder, nonce, isNonceUsed }) => {
+    acc[`${bidder}-${nonce}`] = isNonceUsed
+    return acc
+  }, {} as { [key: string]: boolean })
```
Comment (alpinechicken) [inline utils/auction.ts:78]: Question from my javascript ignorance: do we need to give {} as { [key : string]: boolean} for the initial value of the accumulator or could we just give empty array for the reduce?

Comment (nikkaroraa) [reply utils/auction.ts:78]: here we are dealing with objects, so need to give empty object ie {}. if it'd have been an array, then empty array would have worked

File: utils/auction.ts [hunk 2]
```diff
@@ -65,6 +65,19 @@ export const getUniqueTraders = (bids: Array<Bid>) => {
...
+}
```
Comment (alpinechicken) [inline utils/auction.ts:79]: Why reduce here rather than map again?

File: utils/auction.ts (hunk 1 above)
Comment (alpinechicken) [reply utils/auction.ts:78]: that makes sense but would it work if we just gave it {} like 

```
  return nonceResults.reduce((acc, { bidder, nonce, isNonceUsed }) => {
    acc[`${bidder}-${nonce}`] = isNonceUsed
    return acc
  }, {} as { [key: string]: boolean })
```

**Pull request merged**